ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -388,7 +389,20 @@ void handleNewDeleteEntry(IndexValue deleteValue, IndexValue originalPutValue) {
    */
   void handleNewTtlUpdateEntry(IndexValue ttlUpdateValue, IndexValue originalPutValue) {
     if (recentEntryQueueEnabled) {
-      recentEntryQueue.offer(new Pair<>(ttlUpdateValue, originalPutValue));
+      recentEntryQueue.offer(new RecentQueueItem(ttlUpdateValue, originalPutValue, null));
```

Comment:
I think the passed in `originalPutValue` may not be the actual original put because `updateTtl` method in BlobStore.java doesn't guarantee the found index value is PUT, could be UNDELETE as well. Double check this please.